finetuner-workflow/finetune-workflow.yaml

```diff
@@ -310,53 +336,90 @@ spec:
+            - name: model_path
+              value: "/{{workflow.parameters.pvc}}/finetunes/results-{{workflow.parameters.run_name}}/final"
+        when: "{{workflow.parameters.run_inference}} == true"
+
+    - - name: vllm-inference-service
         template: model-inference-service
         arguments:
           parameters:
+            - name: model_uri
+              value: "vllm/{{workflow.parameters.pvc}}/finetunes/results-{{workflow.parameters.run_name}}/final/vllm/model.tensors"
             - name: model_path
               value: "finetunes/results-{{workflow.parameters.run_name}}/final"
-            - name: model_name
-              value: "final"
         when: "{{workflow.parameters.run_inference}} == true"
   - name: check-model
     inputs:
       parameters:
         - name: model
+        - name: tensorizer_uri
     retryStrategy:
       limit: 1
     container:
-      image: "alpine/curl:3.14"
+      image: "alpine/curl:8.5.0"
       command: [ "/bin/sh", "-c" ]
-      args: [ "/usr/bin/curl -I -XGET -s -o /dev/null -w \"%{http_code}\" https://accel-object.ord1.coreweave.com/tensorized/{{inputs.parameters.model}}/model.tensors | grep -q 200 && echo -n 'true' > /tmp/output.txt || echo -n 'false' > /tmp/output.txt" ]
+      args:
+        - |
+          CHECK_STATUS() {
+            /usr/bin/curl -I -XGET -s -o /dev/null -w '%{http_code}' "$1" | grep -q 200
+          }
+          if [ -n "${TENSORIZER_URI}" ] || CHECK_STATUS "${DEFAULT_URI}"
+          then
+            URI_OUTPUT="${TENSORIZER_URI:-$S3_URI}"
+            OUTPUT='true'
+          else
+            URI_OUTPUT="${TENSORIZER_URI}"
+            OUTPUT='false'
+          fi
+          printf '%s' "${OUTPUT}" > /tmp/output.txt
+          printf '%s' "${URI_OUTPUT}" > /tmp/uri.txt
+      env:
+        - name: TENSORIZER_URI
+          value: "{{inputs.parameters.tensorizer_uri}}"
+        - name: DEFAULT_URI
+          value: "https://object.ord1.coreweave.com/tensorized/{{inputs.parameters.model}}/model.tensors"
+        - name: S3_URI
+          value: "s3://tensorized/{{inputs.parameters.model}}/model.tensors"
     outputs:
       parameters:
         - name: result
           valueFrom:
             path: /tmp/output.txt
+        - name: s3_uri
+          valueFrom:
+            path: /tmp/uri.txt
 
   - name: model-downloader
     inputs:
       parameters:
         - name: model
         - name: dest
         - name: tokenizer_only
+        - name: hf_key
     retryStrategy:
       limit: 1
     container:
       image: "{{workflow.parameters.model_downloader_image}}:{{workflow.parameters.model_downloader_tag}}"
       command: [ "/ko-app/model_downloader" ]
       args: [ "-model", "{{inputs.parameters.model}}",
               "-dest", "{{inputs.parameters.dest}}",
-              "-tokenizer-only", "{{inputs.parameters.tokenizer_only}}" ]
+              "-tokenizer-only={{inputs.parameters.tokenizer_only}}" ]
+      env:
+      - name: HF_API_TOKEN
```

Comment:
Getting deja vu here... this is parameter specified by the model_downloader, yeah? If so then this suggestion is out of scope for this PR, but we should align this variable name with the environment variable used by the huggingface_hub, which is HF_TOKEN. see https://huggingface.co/docs/huggingface_hub/en/package_reference/environment_variables

Reply:
I agree! HF changed the env var name from `HF_API_TOKEN` to `HF_TOKEN` at some point. This change would have to go under our `gpt_bpe` repository.